torch/_dynamo/variables/misc.py

```diff
@@ -374,25 +373,27 @@ def create(target_values, kwargs):
         kwargs.clear()
 
         for key in ["device_type", "dtype", "enabled", "cache_enabled"]:
-            if isinstance(bound_args.arguments[key], VariableTracker):
-                target_values.append(bound_args.arguments[key])
+            arg = bound_args.arguments[key]
+            if isinstance(arg, VariableTracker):
+                if not isinstance(arg, ConstantVariable):
+                    raise unimplemented("autocast with non-constant arguments")
```

Comment:
This shouldn't be a constraint, dynamo wraps ```dtype``` as ```TorchVariable```, which should be supported. You already added [load_const support for ```dtype```](https://github.com/pytorch/pytorch/pull/94137/files#diff-b99ae8eb082046d6353fe4ee536a0d846e1007c7d3b21aefae468ded18fbf13eR703), so just remove this two line.